### PR TITLE
Add FODF reconstruction module.

### PR DESCRIPTION
**Pull request by gagnonanthony**
Not tested yet, will need additional test data.

### CODE REVIEW

File: modules/nf-scil/reconst/fodf/main.nf [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def dwi_shell_tolerance = task.ext.dwi_shell_tolerance ? "--tolerance " + task.ext.dwi_shell_tolerance : ""
+    def fodf_shells = task.ext.fodf_shells ? task.ext.fodf_shells : "\$(cut -d ' ' --output-delimiter=\$'\\n' -f 1- $bval | \
+                                                                    awk -F' ' '{v=int(\$1)}{if(v>=$params.min_fodf_shell_value|| \
+                                                                    v<=$params.b0_thr_extract_b0)print v}' | uniq)"
+    def sh_order = task.ext.sh_order ? "--sh_order " + task.ext.sh_order : ""
+    def sh_basis = task.ext.sh_basis ? "--sh_basis " + task.ext.sh_basis : ""
+    def fa_threshold = task.ext.fa_threshold ? "--fa_t " + task.ext.fa_threshold : ""
+    def md_threshold = task.ext.md_threshold ? "--fa_t " + task.ext.md_threshold : ""
```
Comment (Manonedde) [inline modules/nf-scil/reconst/fodf/main.nf:36]: update --fa_t according to your task.ext.md_threshold

**Review by Manonedde — COMMENTED** [review]
You can run tests with local data. 
Just put in your local path in test_data.config.

**Review by Manonedde — APPROVED** [review]
Everything looks good to me.

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Very nice job ! I have a few minor comments, but this will be ready this ship soon :rocket:

File: tests/modules/nf-scil/reconst/fodf/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        ext.sh_basis = "descoteaux07"
+        ext.fa_threshold = 0.1
+        ext.md_threshold = 0.003
+        ext.fodf_metrics_a_factor = 2.0
+        ext.relative_threshold = 0.1
+        ext.processes = 4
+    }
+
+    withName: "test_reconst_fodf_with_fodf_shells:RECONST_FODF" {
+        ext.fodf_shells = "1000 2000 3000"
```
[the remark this answers is not in the document]
Comment (gagnonanthony) [reply tests/modules/nf-scil/reconst/fodf/nextflow.config:25]: I confirm that the parameters passed through : `` withName: "test_reconst_fodf_with_fodf_shells:RECONST_FODF" `` overwrites the ones passed through `` withName: "RECONST_FODF" ``. 

Echoing the $fa_threshold returned `` --fa_t 0.1 `` for both execution (only passed through `` withName: "RECONST_FODF" `` whereas $fodf_shells returned `` 0 3000 2000 3000 ... `` for the `` withName: "RECONST_FODF" `` and `` 1000 2000 3000 `` for `` test_reconst_fodf_with_fodf_shells:RECONST_FODF ``.

File: modules/nf-scil/reconst/fodf/main.nf [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    scil_compute_ssst_fodf.py dwi_fodf.nii.gz bval_fodf bvec_fodf $frf ${prefix}__fodf.nii.gz \
+        $sh_order $sh_basis --force_b0_threshold \
+        --mask $b0_mask $processes
+    
+    scil_compute_fodf_max_in_ventricles.py ${prefix}__fodf.nii.gz $fa $md \
+        --max_value_output ventricles_fodf_max_value.txt $sh_basis \
+        $fa_threshold $md_threshold -f
+
+    v_max=\$(sed -E 's/([+-]?[0-9.]+)[eE]\\+?(-?)([0-9]+)/(\\1*10^\\2\\3)/g' <<<"\$(cat ventricles_fodf_max_value.txt)")
+    a_threshold=\$(echo "scale=10; $task.ext.fodf_metrics_a_factor * \${v_max}" | bc)
```
Comment (AlexVCaron) [inline modules/nf-scil/reconst/fodf/main.nf:63]: Here there is the possibility `$task.ext.fodf_metrics_a_factor` is null, in which case we should set it with the default from DelAcqua's paper (1.5 or 2 I think, you can check what Tractoflow's doing).

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
One thing, then LGTM 🚢

File: modules/nf-scil/reconst/fodf/main.nf (hunk 3 above)
Comment (gagnonanthony) [reply modules/nf-scil/reconst/fodf/main.nf:63]: Tractoflow is using 2.0 as the default value. Should we hardcode it in the script or still keep it as an optional parameters ?

**Review by AlexVCaron — APPROVED** [review]
All good to go 🚀

**Pull request merged**